### PR TITLE
use HTML for [[SELECTION]] and [[CLIPBOARD]]

### PR DESCRIPTION
**Pull request by SamuelPlentz**
depending on the template type (plain text / html) the selection and clipboard is automatically inserted as plain text / html

### CODE REVIEW

**Comment by SamuelPlentz** [comment]
Resolved the conflict. Any concerns?
One could also call each and every "get_..."-function with the parameter aType. If you want I can change this.

**Comment by jobisoft** [comment]
It has a lot of changes, I need to review them on a rainy weekend.

File: chrome/content/modules/wzQuicktext.jsm [hunk 1]
```diff
@@ -15,25 +15,26 @@ const kFileShortcuts = ['ProfD', 'UsrDocs', 'Home', 'Desk', 'Pers'];
...
+  mDefaultImport:        "",
+  mKeywordKey:           "Tab",
+  mShortcutModifier:     "alt",
+  mShortcutTypeAdv:      false,
+  mQuicktextDir:         null,
+  mObserverList:         [],
+  mOS:                   "WINNT",
+  mCollapseState:        "",
+  mSelectionContent:     "",
+  mSelectionContentHtml: "",
```
Comment (SamuelPlentz) [inline chrome/content/modules/wzQuicktext.jsm:36]: indentation only needed for that new variable `mSelectionContentHtml`

File: chrome/content/modules/wzQuicktextVar.jsm [hunk 2]
```diff
@@ -153,12 +153,18 @@ wzQuicktextVar.prototype = {
           break;
       }
```
Comment (SamuelPlentz) [inline chrome/content/modules/wzQuicktextVar.jsm:155]: The old implementation had a special case for image (do nothing)
The new implementation has 3 special cases (additional parameter aType)
If we add the parameter aType for every "get_..."-function there were no special cases anymore. Perhaps this is even better.

File: chrome/content/modules/wzQuicktextVar.jsm [hunk 3]
```diff
@@ -281,9 +287,14 @@ wzQuicktextVar.prototype = {
     return this.process_file(aVariables);
   }
 ,
-  get_image: function(aVariables)
+  get_image: function(aVariables, aType)
   {
```
Comment (SamuelPlentz) [inline chrome/content/modules/wzQuicktextVar.jsm:291]: Handle the image code depending on aType (html/text) here as it is not done anymore above.

File: chrome/content/modules/wzQuicktextVar.jsm [hunk 4]
```diff
@@ -281,9 +287,14 @@ wzQuicktextVar.prototype = {
     return this.process_file(aVariables);
   }
 ,
```
Comment (SamuelPlentz) [inline chrome/content/modules/wzQuicktextVar.jsm:289]: Handle the image code depending on aType (html/text) here as it is not done anymore above.

**Comment by SamuelPlentz** [comment]
I added some review comments to the code for more clarity. (not normal "//" comments but GIT comments)

**Pull request merged**